handout/features/inheritance.md

```diff
@@ -1,3 +1,84 @@
 ## Inheritance
 
-JavaScript's inheritance works differently from inheritance in other languages, which can be very confusing. ES6 classes provide a syntactic sugar attempting to alleviate the issues with using prototypical inheritance present in ES5. Our recommendation is still to avoid using inheritance or at least deep inheritance hierarchies. Try solving the same problems through delegation instead.
\ No newline at end of file
+JavaScript's inheritance works differently from inheritance in other languages, which can be very confusing. ES6 classes provide a syntactic sugar attempting to alleviate the issues with using prototypical inheritance present in ES5. Our recommendation is still to avoid using inheritance or at least deep inheritance hierarchies. Try solving the same problems through delegation instead.
+
+To illustrate this, let's image we have a zoo application where types of birds are created. In the classical inheritance case, we would define an parent class `Bird` first, then let, say, `Penguin` be a descent of `Bird`:
+```js
+//ES6
+class Bird {
+  constructor(weight, height){
+    this.weight = weight;
+    this.height = height;
+  }
+  walk() {
+    console.log('walk!');
+  }
+}
+
+class Penguin extends Bird {
+  constructor(weight, height){
+    super(weight, height);
+  }
+  swim(){
+    console.log('swim!');
+  }
+}
+
+let penguin = new Penguin(...);
+penguin.walk(); //walk!
+penguin.swim(); //swim!
+```
+and if we use delegation with `class` keyword:
+```js
+//ES6
+class Bird {
+  constructor(weight, height){
+    this.weight = weight;
+    this.height = height;
+  }
+  walk() {
+    console.log('walk!');
+  }
+}
+
+class Penguin {
+  constructor(weight, height){
+    this.bird = new Bird(weight, height);
+  }
+  walk() {
+    this.bird.walk();
+  }
+  swim() {
+    console.log('swim!');
+  }
+}
+
+let penguin = new Penguin(...);
+penguin.walk(); //walk!
+penguin.swim();  //swim!
+```
```

Comment:
I think here we should create the delegator instance before (Bird) and then pass it to the Penguin constructor. It will look similar to the reader to the DI.

```js
class Penguin {
  constructor(bird) {
    this.bird = bird;
  }
  walk() {
    this.bird.walk();
  }
  swim() {
    console.log('swim!');
  }
}

let bird = new Bird(...);
let penguin = new Penguin(bird);

penguin.walk(); //walk!
penguin.swim();  //swim!
```